### PR TITLE
bump up version for rocm-4.1. release

### PR DESCRIPTION
**Pull request by srekolam**
this is done to address the new rocm-4.1.0 release

### CODE REVIEW

File: var/spack/repos/builtin/packages/llvm-amdgpu/package.py [hunk 1]
```diff
@@ -32,6 +33,7 @@ class LlvmAmdgpu(CMakePackage):
     depends_on('python', type='build')
     depends_on('z3', type='link')
     depends_on('zlib', type='link')
+    depends_on('unzip', type='link')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/llvm-amdgpu/package.py]:
```suggestion
```
unzip is a binary, not sure what it does here? can you comment on this?

if i grep `unzip` in the sources I can only find a makefile in the polly subproject which is not built by us?

File: var/spack/repos/builtin/packages/hip-rocclr/package.py [hunk 2]
```diff
@@ -92,6 +94,14 @@ def url_for_version(self, version):
              placement='opencl-on-vdi',
              when='@4.0.0')
 
+    resource(name='opencl-on-vdi',
```
Comment (haampie) [inline var/spack/repos/builtin/packages/hip-rocclr/package.py:97]: Any news on moving this to a separate package? I'd like to use libOpenGL.so too :upside_down_face:

File: var/spack/repos/builtin/packages/llvm-amdgpu/package.py [hunk 3]
```diff
@@ -40,7 +42,7 @@ class LlvmAmdgpu(CMakePackage):
 
     # Will likely only be fixed in LLVM 12 upstream
     patch('fix-system-zlib-ncurses.patch', when='@3.5.0:3.8.0')
-    patch('fix-ncurses-3.9.0.patch', when='@3.9.0:')
+    patch('fix-ncurses-3.9.0.patch', when='@3.9.0:4.0.0')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/llvm-amdgpu/package.py:44]: Yay, happy to see we can drop this patch now :)

**Review by haampie — CHANGES_REQUESTED** [review]
Overall looks good, but I don't see where `unzip` is used in llvm.

**Comment by srekolam** [comment]
llvm-amdgpu has dependency on Z3 which again depends on py-setuptools . Before installing py-setuptools@50.3.2, it checks for unzip package and for this , i used to install and load unzip in my centos docker. i thought it is better to include unzip  in the llvm-amdgpu package. hence i added that change.

File: var/spack/repos/builtin/packages/hip-rocclr/package.py (hunk 2 above)
Comment (srekolam) [reply var/spack/repos/builtin/packages/hip-rocclr/package.py:97]: i have started on this package and i see a couple of build errors which require some fixes. I will fix them and upstream it soon.

**Comment by haampie** [comment]
Hm ok, you'll have to add dependencies to the actual dependent package, because right now neither z3 nor py-setuptools would have unzip available.

File: var/spack/repos/builtin/packages/llvm-amdgpu/package.py (hunk 1 above)
Comment (srekolam) [reply var/spack/repos/builtin/packages/llvm-amdgpu/package.py]: i have made the changes per your suggestions.

**Review by haampie — APPROVED** [review]

**Pull request merged**